tests/system-tests/rdscore/internal/rdscorecommon/sriov-pod-level-bond.go

```diff
@@ -0,0 +1,1114 @@
+	}
+
+	cmdToRun := []string{"bash", "-c",
+		fmt.Sprintf("testcmd -protocol tcp -port %s -interface net3 -packages %s -timeoutTCP %s -server %s",
+			serverPort, packetsNumber, timeout, serverIPAddr)}
+
+	glog.V(100).Infof("Execute command: %q", cmdToRun)
+
+	var output string
+
+	err = wait.PollUntilContextTimeout(
+		context.TODO(),
+		time.Second*3,
+		time.Minute,
+		true,
+		func(ctx context.Context) (bool, error) {
+			result, err := clientPod.ExecCommand(cmdToRun, clientPod.Object.Spec.Containers[0].Name)
+
+			if err != nil {
+				glog.V(100).Infof("Error running command from within a pod %q: %v",
+					clientPod.Object.Name, err)
+
+				return false, nil
+			}
+
+			glog.V(100).Infof("Successfully executed command from within a pod %q in namespace %q",
+				clientPod.Object.Name, clientPod.Definition.Namespace)
+
+			output = result.String()
+			glog.V(100).Infof("Command's output:\n\t%v", output)
+
+			return true, nil
+		})
+
+	if err != nil {
+		return "", fmt.Errorf("failed to run command from within pod %s: %w", clientPod.Object.Name, err)
+	}
+
+	return output, nil
+}
+
+func findInCmdExecOutput(cmdExecOutput, stringToFind string) (bool, error) {
+	var err error
+
+	matchesFound := 0
+
+	if len(cmdExecOutput) != 0 {
+		buf := new(bytes.Buffer)
+		_, err = buf.WriteString(cmdExecOutput)
+
+		stringToFind, _ := regexp.Compile(stringToFind)
+
+		if err != nil {
+			glog.V(100).Infof("error in copying info from the cmdExecOutput to buffer: %v", err)
+
+			return false, fmt.Errorf("error in copying info from the cmdExecOutput to buffer: %w", err)
+		}
+
+		scanner := bufio.NewScanner(buf)
+
+		for scanner.Scan() {
+			logLine := scanner.Text()
+			if stringToFind.MatchString(logLine) {
+				matchesFound++
+			}
+		}
+
+		if matchesFound < 1 {
+			glog.V(100).Infof("Expected string %q not found in the output: %s", stringToFind, cmdExecOutput)
+
+			return false, fmt.Errorf("expected string %q not found in the output : %s", stringToFind, cmdExecOutput)
+		}
+	}
+
+	return true, nil
+}
+
+func scanClientPodTrafficOutput(clientPodOutput string) (bool, error) {
+	glog.V(100).Infof("client pod output: %s", clientPodOutput)
+
+	isFound, err := findInCmdExecOutput(clientPodOutput, tcpTestPassedMsg)
+
+	if err != nil {
+		glog.V(100).Infof("Failed to parse clientPodOutput due to %v", err)
+
+		return false, fmt.Errorf("failed to parse clientPodOutput due to %w", err)
+	}
+
+	if !isFound {
+		glog.V(100).Infof("TCP traffic transmission failure detected: %s", clientPodOutput)
+
+		return false, fmt.Errorf("tcp traffic transmission failure detected: %s", clientPodOutput)
+	}
+
+	return true, nil
+}
+
+func getBondActiveInterface(clientPod *pod.Builder) (string, error) {
+	glog.V(90).Infof("Getting bond active VF interface for the pod %s in namespace %s",
+		clientPod.Definition.Name, clientPod.Definition.Namespace)
+
+	var output bytes.Buffer
+
+	var result string
+
+	var err error
```

Comment:
nit, this could be grouped like:
```suggestion
	var (
	    output bytes.Buffer
            result   string
            err        error
         )
```